bangazonapi/views/order.py

```diff
@@ -147,11 +147,11 @@ def list(self, request):
             ]
         """
         customer = Customer.objects.get(user=request.auth.user)
-        orders = Order.objects.filter(customer=customer).exclude(payment_type_id=None)
+        orders = Order.objects.filter(customer=customer)
 
-        payment = self.request.query_params.get('payment_id', None)
+        payment = self.request.query_params.get('payment_type', None)
         if payment is not None:
-            orders = orders.filter(payment__id=payment)
+            orders = orders.filter(payment_type_id=payment)
```

Comment:
Again, validate the `payment_type_id` using `get()` so you can filter with objects, not integers

Reply:
done